share/completions/npm.fish

```diff
@@ -150,58 +182,87 @@ complete -f -c npm -n '__fish_npm_using_command owner' -a ls -d 'List package ow
 complete -f -c npm -n '__fish_npm_using_command owner' -a add -d 'Add a new owner to package'
 complete -f -c npm -n '__fish_npm_using_command owner' -a rm -d 'Remove an owner from package'
 
+# pack
+complete -f -c npm -n __fish_npm_needs_command -a pack -d 'Create a tarball from a package'
+complete -f -c npm -n '__fish_npm_using_command pack' -l dry-run -d 'Only report what it would have done'
+complete -f -c npm -n '__fish_npm_using_command pack' -l json -d 'Output JSON data'
+complete -c npm -n '__fish_npm_using_command pack' -l pack-destination -d 'Tarball destination directory'
+
+# publish
+complete -f -c npm -n __fish_npm_needs_command -a publish -d 'Publish a package'
+complete -x -c npm -n '__fish_npm_using_command publish' -l tag -d 'Upload to tag'
+complete -x -c npm -n '__fish_npm_using_command publish' -l access -d 'Restrict access' -a "public\t'Publicly viewable' restricted\t'Restricted access (scoped packages only)'"
+complete -f -c npm -n '__fish_npm_using_command publish' -l dry-run -d 'Only report what it would have done'
+complete -x -c npm -n '__fish_npm_using_command publish' -l otp -d '2FA one-time password'
+## todo provenance ?
+
 # remove
-for c in r remove rm un uninstall unlink
-    complete -f -c npm -n __fish_npm_needs_command -a "$c" -d 'remove package' -xa '(__yarn_installed_packages)'
-    complete -x -c npm -n "__fish_npm_using_command $c" -s g -l global -d 'remove global package'
+for c in uninstall unlink remove rm r un
+    complete -f -c npm -n __fish_npm_needs_command -a "$c" -d 'Remove a package'
+    complete -x -c npm -n "__fish_npm_using_command $c" -d 'remove package' -a '(__yarn_installed_local_packages)'
```

Comment:
Hold on, is it right that this npm completion tries the yarn installed packages?

Reply:
I noticed you didn't add it, I just thought it was suspect and might be a mistake.

Fine to leave it as-is.

Reply:
should i rename it (and all its derivatives) to `__yarn_`, it probably makes more sense for yarn to be using npm functions than the other way around?

Reply:
Yeah, seems right.